DZ Robot/src/by/teachmeskills/robot/Robot.java

```diff
@@ -15,7 +23,27 @@ public Robot(IHead head, IHand hand, ILeg leg) {
         this.leg = leg;
     }
 
-    public Robot(){}
+    public Robot() {
+    }
+
+    public Robot(SonyHand sonyHand, SonyHead sonyHead, ToshibaLeg toshibaLeg) {
+        this.hand = sonyHand;
+        this.head = sonyHead;
+        this.leg = toshibaLeg;
+
+    }
+
+    public Robot(ToshibaHand toshibaHand, ToshibaHead toshibaHead, SonyLeg sonyLeg) {
+        this.hand = toshibaHand;
+        this.head = toshibaHead;
+        this.leg = sonyLeg;
+    }
+
+    public Robot(SamsungHand samsungHand, SamsungHead samsungHead, SonyLeg sonyLeg) {
+        this.hand = samsungHand;
+        this.head = samsungHead;
+        this.leg = sonyLeg;
+    }
```

Comment:
you don't need 3 constructors
polymorphism allows you to have 1 constructor with 3 params
public Robot(IHand hand, IHead head, ILeg leg)
and then you can call it with all possible combinations of implementations of robot parts